test/Test/Iris/Cli.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+module Test.Iris.Cli (cliSpec) where
+
+import Test.Hspec (Spec, describe, it)
+
+import Iris.Cli.ParserInfo (cmdParserInfo)
+import Iris.Settings (defaultCliEnvSettings, cliEnvSettingsVersionSettings)
+import Iris.Cli.Cmd (Cmd (..))
+import Iris.Cli.Version (defaultVersionSettings)
+import qualified Options.Applicative as Opt
+import Data.List (isInfixOf)
+import qualified Paths_iris as Autogen
+
+
+
+type HelpOptions = [String]
+
+cliSpec :: Spec
+cliSpec = describe "Cli Options" $ do
+    let parserPrefs :: Opt.ParserPrefs = Opt.defaultPrefs 
+    it "help without version environment should output the options help and no-input and should not output the options version and numeric-version" $ do
+        let parserInfo :: Opt.ParserInfo (Cmd ()) = cmdParserInfo defaultCliEnvSettings
+        let result :: Opt.ParserResult (Cmd ()) = Opt.execParserPure parserPrefs parserInfo ["--help"]
+        parseResultHandler result (\text -> helpHasOptions text ["--help", "--no-input"] && not (helpHasOptions text ["--version", "--numeric-version"]))
```

Comment:
I'm thinking about writing this test differently. I propose to use the **Golden testing** approach for this function when we describe the entire `--help` output as a hardcoded string and then compare it with the result.

This way, we can always notice all the changes in the `--help` whenever we introduce new behaviour.

I haven't compiler the below code but I imagine something like this, using [functions from `hspec`](https://hackage.haskell.org/package/hspec-expectations-0.8.2/docs/Test-Hspec-Expectations.html).

```haskell
case result of
    Opt.Failure (Opt.ParserFailure getFailure) -> do
        let (helpText, _exitCode, _int) = getFailure "<iris-test>"
        show helpText `shouldBe` expectedHelpText
    other -> expectationFailure $ "Expected 'Failure' but got: " <> show other
```

Where `expectedHelpText` is defined as a top-level value like:

```haskell
expectedHelpText :: String
expectedHelpText =
"Usage: <iris-test> [--no-input]\n\
\...
"
```

You can use Haskell syntax for multiline strings or use `unlines` and a list of strings. Whatever you find more pleasing, I'm fine with both 🙂

Reply:
Ah I had not heard this term before. I will implement it.

Reply:
So the `show other` gives 
```
• No instance for (Show (Cmd ())) arising from a use of ‘show’
• In the second argument of ‘(<>)’, namely ‘show other’
  In the second argument of ‘($)’, namely
    ‘"Expected 'Failure' but got: " <> show other’
  In the expression:
    expectationFailure $ "Expected 'Failure' but got: " <> show othertypecheck(-Wdeferred-type-errors)
```
So it requires implement show for Cmd.

Reply:
I have not implemented a show instance for Cmd, but other than that I think I have completed your requests

Reply:
Would it make sense to do something like
```
instance (Show a) => Show (Cmd a) where
    show Cmd{..} = show cmdCmd
```?